src/fields.cpp

```diff
@@ -213,6 +213,44 @@ fields_chunk::~fields_chunk() {
   if (new_s && new_s->refcount-- <= 1) delete new_s; // delete if not shared
 }
 
+static void split_into_tiles(grid_volume gvol, std::vector<grid_volume> &result) {
+  const size_t base_nowned_min = 1000000;
+  if (gvol.nowned_min() < base_nowned_min) {
+    result.push_back(gvol);
+    return;
+  }
+
+  int best_split_point;
+  direction best_split_direction;
+  double left_effort_fraction;
+  gvol.find_best_split(2, false /* fragment_cost */, best_split_point, best_split_direction,
+                       left_effort_fraction);
+  grid_volume left_gvol = gvol.split_at_fraction(false, best_split_point, best_split_direction);
+  split_into_tiles(left_gvol, result);
+  grid_volume right_gvol = gvol.split_at_fraction(true, best_split_point, best_split_direction);
+  split_into_tiles(right_gvol, result);
+  return;
+}
+
+// First check that the tile volumes gvs do not intersect and that they add
+// up to the chunk's total grid_volume gv
+static void check_tiles(grid_volume gv, std::vector<grid_volume> gvs) {
```

Comment:
As another check, the `little_corner()` of each grid volume should consist purely of *even* numbers.  (This should be enforced by `split_at_fraction`.)

Reply:
Pass by const ref

Reply:
Fixed.